### PR TITLE
DSE stopgap fixes

### PR DESCRIPTION
**Pull request by mlange05**
A small set of fixes that circumvent DSE failures some corner cases laid bare by the current seismic operators. These are meant as an intermediate steps to fix issue #218 while @FabioLuporini finishes the full engine re-write.

### CODE REVIEW

**Comment by mloubout** [comment]
TRAVIS looks happy, I'll have a deeper look at it, but from the changes it seems to be ok.

File: examples/acoustic/acoustic_example.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def run(dimensions=(50, 50, 50), spacing=(20.0, 20.0, 20.0), tn=1000.0,
     info("Applying Adjoint")
     Acoustic.Adjoint(rec, dse=dse, dle=dle, legacy=legacy)
     info("Applying Born")
-    Acoustic.Born(dm, dse=None, dle=dle, legacy=legacy)
+    Acoustic.Born(dm, dse=dse, dle=dle, legacy=legacy)
```
Comment (mloubout) [inline examples/acoustic/acoustic_example.py]: Needs #233  to be merged first as this part is not tested with current setup. My fault, I removed the `full_run` t some point locally and forgot to put it back.

Comment (mlange05) [inline examples/acoustic/acoustic_example.py]: Sure, #233 goes in first..

**Comment by mloubout** [comment]
Ok this looks good and can me merged in my opinion (`acoustic_example runs everything fine). What's left to do (other PR) is making sure that multi-stencil kernels like `Born` use `openmp` for all the stencils, not just the first encountered.

**Comment by FabioLuporini** [comment]
I think this can be merged; I'll do the rebase once my branch is ready

**Comment by mlange05** [comment]
Ok, thanks. Merging...

**Pull request merged**